pythonwhat/checks/check_wrappers.py

```diff
@@ -628,62 +627,131 @@ def __init__(self, i):
             - ``check_body()`` zooms in on the body of the ``while`` loop, and ``has_equal_value()`` checks whether rerunning this body
               updates ``i`` as expected when ``i`` is temporarily set to 3 with ``set_env()``.
 
-        """
+        """,
     },
-    'with': {
-        'typestr': '{{ordinal}} `with` statement',
-        'docstr': """Check whether a with statement was coded zoom in on it.
+    "with": {
+        "typestr": "{{ordinal}} `with` statement",
+        "docstr": """Check whether a with statement was coded zoom in on it.
 
         Args:
             index: the index of the``with`` statement to verify (0-based)
             {{typestr}}
             {{missing_msg}}
             {{expand_msg}}
 
-        """
+        """,
     },
 }
 
-scts = {}
+scts = dict()
 
 # make has_equal_part wrappers
 
-scts['has_equal_name'] = partial(has_equal_part, 'name', msg='Make sure to use the correct {{name}}, was expecting {{sol_part[name]}}, instead got {{stu_part[name]}}.')
-scts['is_default'] = partial(has_equal_part, 'is_default', msg="Make sure it {{ 'has' if sol_part.is_default else 'does not have'}} a default argument.")
+
+# todo: check @wraps implementation
+
+# def partial_with_state(func, *partial_args, **partial_kwargs):
+#     @wraps(func)
+#     def state_partial(state, *args, **kwargs):
+#         func(state, *partial_args, *args, **{**partial_kwargs, **kwargs})
+#     return state_partial
+
+
+# def right_args_partial(func, *last_args, **partial_kwargs):
+#     kwargs_partial = partial(func, **partial_kwargs)
+#
+#     @wraps(func)
+#     def full_partial(*first_args, **kwargs):
+#         kwargs_partial(*first_args, *last_args, **kwargs)
+#
+#     return full_partial
+
+
+def partial_with_offset(func, *partial_args, offset=1, **partial_kwargs):
```

Comment:
`offset` seems like an arg that could conflict. You can curry it up:

```python3
def partial_with_offset(offset = 1):
  def bound_partial_with_offset(func, *partial_args, **partial_kwargs):
    ...
  return bound_partial_with_offset
```
If u want to avoid having to change a lot you can name the above `generic_partial_with_offset` and 

```python3
partial_with_offset = generic_partial_with_offset(offset = 1)
```